Homework/homework4/task1.py

```diff
@@ -0,0 +1,51 @@
+"""
+Write a function that gets file path as an argument.
+
+Read the first line of the file.
+If first line is a number return true if number in an interval [1, 3)*
+and false otherwise.
+In case of any error, a ValueError should be thrown.
+Write a test for that function using pytest library.
+Definition of done:
+ - function is created
+ - function is properly formatted
+ - function has positive and negative tests
+ - all temporary files are removed after test run
+You will learn:
+ - how to test Exceptional cases
+ - how to clean up after tests
+ - how to check if file exists**
+ - how to handle*** and raise**** exceptions in test. Use sample from the documentation.
+* https://en.wikipedia.org/wiki/Interval_(mathematics)#Terminology
+** https://docs.python.org/3/library/os.path.html
+*** https://docs.python.org/3/tutorial/errors.html#handling-exceptions
+**** https://docs.python.org/3/tutorial/errors.html#raising-exceptions
+"""
+import os
+
+
+def is_number(line: str) -> int or float:
+    """Check if the string is a number."""
+    if line.isdigit():
+        return int(line)
+    else:
+        try:
+            return float(line)
+        except ValueError:
+            raise ValueError("Not a number")
+
+
+def is_in_interval(num: int or float) -> bool:
```

Comment:
you can implement this function as a single line.  And it will be more readable.